File: spring-boot-template/src/main/java/tui/meta/challenge/quotes/controller/model/QuoteDto.java
package tui.meta.challenge.quotes.controller.model;

import tui.meta.challenge.quotes.repository.model.QuoteEntity;

public record QuoteDto(
    String id,
    String text,
    String author,
    String genre) {

    public QuoteDto(QuoteEntity entity) {
        this(entity.getId(), entity.getText(), entity.getAuthor(), entity.getGenre());
    }
}

File: micronaut-template/src/main/kotlin/tui/meta/challenge/controller/model/QuoteDto.kt
package tui.meta.challenge.controller.model

import io.micronaut.serde.annotation.Serdeable
import tui.meta.challenge.repository.model.QuoteEntity

@Serdeable
data class QuoteDto(
    val id: String,
    val text: String,
    val author: String,
    val genre: String
) {
    constructor(entity: QuoteEntity) : this(
        id = entity.id,
        text = entity.quoteText,
        author = entity.quoteAuthor,
        genre = entity.quoteGenre
    )
}